homeassistant/components/vicare/sensor.py

```diff
@@ -743,9 +825,9 @@ def update(self) -> None:
                 if self.entity_description.unit_getter:
                     vicare_unit = self.entity_description.unit_getter(self._api)
                     if vicare_unit is not None:
-                        self._attr_device_class = VICARE_UNIT_TO_DEVICE_CLASS.get(
-                            vicare_unit
-                        )
+                        device_class = VICARE_UNIT_TO_DEVICE_CLASS.get(vicare_unit)
```

Comment:
Please take in account, this is stuff in the try block that doesn't raise, If you could move it out, that'd be great

Reply:
This is based on `self.entity_description.unit_getter` which can raise an exception.

Reply:
Done